apps/src/lib/client/tx.rs

```diff
@@ -975,6 +976,90 @@ pub async fn submit_withdraw(ctx: Context, args: args::Withdraw) {
     process_tx(ctx, &args.tx, tx, Some(default_signer)).await;
 }
 
+pub async fn submit_validator_commission_change(
+    ctx: Context,
+    args: args::TxCommissionRateChange,
+) {
+    let epoch = rpc::query_epoch(args::Query {
+        ledger_address: args.tx.ledger_address.clone(),
+    })
+    .await;
+
+    let tx_code = ctx.read_wasm(TX_CHANGE_COMMISSION_WASM);
+    let client = HttpClient::new(args.tx.ledger_address.clone()).unwrap();
+
+    // Check that the submitter of the tx is a validator
+    match ctx.wallet.get_validator_data() {
```

Comment:
This works, but I think it would be little better to use `let validator = ctx.get(&args.validator); if rpc::is_validator(&validator) { ...` instead to respect the provided argument. In vast majority of cases (if not all?) they will be the same address though

Reply:
Ok, yeah I was also second-guessing whether we even want `validator` to be an arg in `args::TxCommissionRateChange` since in all cases, this address should be the same as the one of the tx submitter. I will change, though.